pkg/controllers/cluster/cluster_controller.go

```diff
@@ -543,6 +543,7 @@ func (c *Controller) tryUpdateClusterHealth(ctx context.Context, cluster *cluste
 					currentCondition.Reason = "ClusterStatusUnknown"
 					currentCondition.Message = "Cluster status controller stopped posting cluster status."
 					currentCondition.LastTransitionTime = nowTimestamp
+					meta.SetStatusCondition(&cluster.Status.Conditions, *currentCondition)
```

Comment:
I think the condition of `cluster` could be updated by the pointer(`currentCondition`), so we might don't need to set the conditions explicitly.

cc author @Garrybest to confirm.

Reply:
> currentCondition := meta.FindStatusCondition([]metav1.Condition{initCondition}, "Ready")

I guess the `[]metav1.Condition{initCondition}` just builds a new slice, which means `currentCondition` essentially points to a copy of `initCondition`. So changes made on `currentCondition` won't reflect to `initCondition`.

Reply:
Could you please try the following sample? 
```golang
package main

import (
	"encoding/json"
	"fmt"

	"k8s.io/apimachinery/pkg/api/meta"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

func main() {
	initCondition := metav1.Condition{
		Type:               "Ready",
		Status:             metav1.ConditionTrue,
		ObservedGeneration: 123,
		Reason:             "Reason",
		Message:            "Message",
	}
        conditions := []metav1.Condition{initCondition} // build a new slice to simulate 'cluster.Status.Conditions'
	currentCondition := meta.FindStatusCondition(conditions, "Ready") // simulates meta.FindStatusCondition(cluster.Status.Conditions, clusterConditionType)
	currentCondition.Status = metav1.ConditionUnknown
	currentCondition.Reason = "ClusterStatusUnknown"
	currentCondition.Message = "Cluster status controller stopped posting cluster status."

	newData, _ := json.MarshalIndent(currentCondition, " ", " ")
	fmt.Printf("newData: %v\n\n", string(newData))

	oldData, _ := json.MarshalIndent(conditions, " ", " ") // dump conditions, the old conditions should be updated as well
	fmt.Printf("oldData: %v\n", string(oldData))
}
```

Reply:
@RainbowMango from the program output to see the code is no problem:)

program output:
```
	newData: {
	  "type": "Ready",
	  "status": "Unknown",
	  "observedGeneration": 123,
	  "lastTransitionTime": null,
	  "reason": "ClusterStatusUnknown",
	  "message": "Cluster status controller stopped posting cluster status."
	 }
	
	oldData: [
	  {
	   "type": "Ready",
	   "status": "Unknown",
	   "observedGeneration": 123,
	   "lastTransitionTime": null,
	   "reason": "ClusterStatusUnknown",
	   "message": "Cluster status controller stopped posting cluster status."
	  }
	 ]
```